app/controllers/ingredients_controller.rb

```diff
@@ -1,18 +1,18 @@
 class IngredientsController < ApplicationController
   def index
-    
     @ingredients = Ingredient.all
   end
 
   def create
-  @recipe = Recipe.find(params[:recipe_id])
-  @ingredient = @recipe.ingredients.new(ingredient_params)
-  @ingredient.user = current_user
-  if @ingredient.save
-    flash[:notice] = "ingredient Added."
-    redirect_to recipe_path(@recipe)
-  else
-    flash[:notice] = @ingredient.errors.full_messages.join("! ")
-    render 'recipes/show'
+    @recipe = Recipe.find(params[:recipe_id])
+    @ingredient = @recipe.ingredients.new(ingredient_params)
+    @ingredient.user = current_user
+    if @ingredient.save
+      flash[:notice] = "ingredient Added."
+      redirect_to recipe_path(@recipe)
+    else
+      flash[:notice] = @ingredient.errors.full_messages.join("! ")
```

Comment:
Let's keep errors like this in the view, at the top of your form. No need to assign these to the flash. 

In your view:

```erb
<% if !@ingredient.errors.full_messages.empty? %>
  <ul>  
    <% @ingredient.errors.full_messages.each do |error| %>
      <li><%= error %></li>
    <% end %>
  </ul>
<% end %>  
```